Algorithms_on_strings_Shestakov

Greenprize opened the pull request.

Greenprize committed:
Algorithms_on_strings_Shestakov

Nastasia8 commented on Algorithms_on_strings.py:
```diff
@@ -0,0 +1,107 @@
+            p[i + 1] = 0
+        if p[i + 1] < 2:
+            kek = i + 1
+    return kek 
```
3 и 4 задачи реализованы по Вашей идее?)

Greenprize replied:
На форумах 16 года примерно так описывают реализацию

Greenprize committed:
Sorting options

4 задание отваливается на 19 тесте, а также я всё-таки вспомнил про обновление 7-го

Greenprize committed:
Data_structures_stack__queue__dec__linked_lists_Shestakov

Так, я осознаю, как сделать 3-е задание, чтобы на это уходило 2O(n) времени, но я туп

Nastasia8 commented on Data_structures_stack__queue__dec__linked_lists.py:
```diff
@@ -0,0 +1,38 @@
+
+
+#Задание 2
+from collections import deque               #если вы хотите проверить меня на знание кода, то просто напише, где комменты оставить
```
Ну, мы ведь эту задачу вместе делали)

Nastasia8 commented on Data_structures_stack__queue__dec__linked_lists.py:
```diff
@@ -0,0 +1,38 @@
+    if i == ")":                             #это проверка на соотвествие элементов
+        if not stack or stack[-1] != "(":    #если стек пустой или последний элемент стека не ")"
+            count += 1                       #счётчик сделал +1
+            continue                         #продолжаем
```
#продолжаем - плохой комментарий)

Nastasia8 commented on Data_structures_stack__queue__dec__linked_lists.py:
```diff
@@ -0,0 +1,38 @@
+    else:
+        stack.append(i)          #добавляем в стек (УРА, в нём что-то появилось)
+
+print(count+len(stack))          #выводим (больше этого не повториться, а то мне надоело)
```
:)
